src/App.tsx

```diff
@@ -1,14 +1,75 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { getTodos, getUser } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [query, setQuery] = useState<string>('');
+  const [status, setStatus] = useState<'all' | 'active' | 'completed'>('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isModalLoading, setIsModalLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(data => {
+        setTodos(data);
+        setFilteredTodos(data);
+      })
+      .catch(error => {
+        /* eslint-disable no-console */
+        console.error('Failed to fetch todos:', error);
+      })
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  useEffect(() => {
+    let result = todos;
+
+    if (status !== 'all') {
+      const isCompleted = status === 'completed';
+
+      result = result.filter(todo => todo.completed === isCompleted);
+    }
+
+    if (query) {
+      result = result.filter(todo =>
+        todo.title.toLowerCase().includes(query.toLowerCase()),
+      );
+    }
+
+    setFilteredTodos(result);
+  }, [query, status, todos]);
+
+  const handleShowModal = (todo: Todo) => {
+    setIsModalLoading(true);
+    setSelectedTodo(todo);
+
+    getUser(todo.userId)
+      .then(user => setSelectedUser(user))
+      .catch(error => {
+        console.error('Failed to fetch user:', error)
```

Comment:
Similar to the previous comment, consider providing user feedback when an error occurs in fetching user data.